components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/uaepass/authenticator/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,1109 @@
+import org.wso2.carbon.identity.core.util.IdentityCoreConstants;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.uaepass.authenticator.exception.UAEPassAuthnFailedException;
+import org.wso2.carbon.identity.uaepass.authenticator.exception.UAEPassUserInfoFailedException;
+import org.wso2.carbon.identity.uaepass.authenticator.internal.UAEPassDataHolder;
+import org.wso2.carbon.user.api.UserRealm;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.text.ParseException;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.HashSet;
+
+public class UAEPassAuthenticator extends AbstractApplicationAuthenticator
+        implements FederatedApplicationAuthenticator {
+
+    private static final Log LOG = LogFactory.getLog(UAEPassAuthenticator.class);
+
+    /**
+     * Checks wheather the request and response can handle by the federated IdP.
+     *
+     * @param request
+     * @return Boolean
+     */
+    @Override
+    public boolean canHandle(HttpServletRequest request) {
+
+        return UAEPassAuthenticatorConstants.UAE.LOGIN_TYPE.equals(getLoginType(request));
+    }
+
+
+    /**
+     * Returns the federated IdP component's friendly name.
+     *
+     * @return String
+     */
+    @Override
+    public String getFriendlyName() {
+        return UAEPassAuthenticatorConstants.UAE.FEDERATED_IDP_COMPONENT_FRIENDLY_NAME;
+    }
+
+
+    /**
+     * Returns the federated IdP component name.
+     *
+     * @return String
+     */
+    @Override
+    public String getName() {
+        return UAEPassAuthenticatorConstants.UAE.FEDERATED_IDP_COMPONENT_NAME;
+    }
+
+
+    /**
+     * Returns the claim dialect URL.
+     *
+     * @return String
+     */
+    @Override
+    public String getClaimDialectURI() {
+        return UAEPassAuthenticatorConstants.UAEPassRuntimeConstants.OIDC_DIALECT;
+    }
+
+
+    /**
+     * Returns an unique string to identify each request and response separetely.
+     *
+     * @param request
+     * @return String
+     */
+    @Override
+    public String getContextIdentifier(HttpServletRequest request) {
+
+        String state = request.getParameter(UAEPassAuthenticatorConstants.UAE.OAUTH2_PARAM_STATE);
+        if (StringUtils.isNotBlank(state)) {
+            return state.split(",")[0];
+        } else {
+            LOG.error("An unique identifier couldn't issue for both Request and Response. ContextIdentifier is NULL");
```

Comment:
An unique identifier couldn't issue for both Request and Response-> An unique identifier cannot be issued for both Request and Response.
Full stop is missing in second sentence.